fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java

```diff
@@ -0,0 +1,100 @@
+import com.starrocks.sql.optimizer.operator.logical.LogicalAggregationOperator;
+import com.starrocks.sql.optimizer.operator.logical.LogicalScanOperator;
+import com.starrocks.sql.optimizer.operator.pattern.Pattern;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.rule.RuleType;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+// optimization for select min/max/count distinct dt from table where predicates
+// and predicates only has partition columns.
+
+public class PartitionColumnValueOnlyOnScanRule extends TransformationRule {
+    public PartitionColumnValueOnlyOnScanRule() {
+        // agg -> project -> scan[checked in `check`]
+        super(RuleType.TF_REWRITE_PARTITION_COLUMN_ONLY_AGG,
+                Pattern.create(OperatorType.LOGICAL_AGGR).addChildren(Pattern.create(OperatorType.LOGICAL_PROJECT)));
+    }
+
+    @Override
+    public boolean check(final OptExpression input, OptimizerContext context) {
+        if (!context.getSessionVariable().isEnablePartitionColumnValueOnlyOptimization()) {
+            return false;
+        }
+
+        LogicalAggregationOperator aggregationOperator = (LogicalAggregationOperator) input.getOp();
+        Operator operator = input.getInputs().get(0).getInputs().get(0).getOp();
+        if (!(operator instanceof LogicalScanOperator)) {
+            return false;
+        }
+        LogicalScanOperator scanOperator = (LogicalScanOperator) operator;
+
+        // we can only apply this rule to the queries met all the following conditions:
+        // 1. only contain MIN/MAX/COUNT DISTINCT agg functions
+        // 2. scan operator only output partition columns.(right now we only support 1 column)
+        if (!allPartitionColumnInScanOperator(scanOperator)) {
+            return false;
+        }
+
+        boolean allValid = aggregationOperator.getAggregations().values().stream().allMatch(aggregator -> {
+            AggregateFunction aggregateFunction = (AggregateFunction) aggregator.getFunction();
+            String functionName = aggregateFunction.functionName();
+
+            // min/max/count distinct(a)
+            if (!(functionName.equals(FunctionSet.MAX) || functionName.equals(FunctionSet.MIN) ||
+                    (functionName.equals(FunctionSet.COUNT) && aggregator.isDistinct()))) {
+                return false;
+            }
+
+            return true;
+        });
+        return allValid;
+    }
+
+    private static boolean allPartitionColumnInScanOperator(LogicalScanOperator scanOperator) {
+        int pc = 0;
+        Set<String> partitionColumns = scanOperator.getPartitionColumns();
+        for (ColumnRefOperator c : scanOperator.getOutputColumns()) {
+            if (!partitionColumns.contains(c.getName())) {
+                // materialized column could be taken durding prune stage.
+                if (!scanOperator.getScanOptimzeOption().getCanUseAnyColumn()) {
```

Comment:
select count(1) from tbl where partition_col = 'xxx' will also use this rule to optimize?